lib/table.c

```diff
@@ -2260,6 +2260,17 @@ simplifier_merge_ancestors(simplifier_t *self, node_id_t input_id)
         goto out;
     }
 
+    /* check if the input ID is a sample */
+    output_id = self->node_id_map[input_id];
+    coalescence = sample_node = self->is_sample[input_id];
+    if (sample_node) {
+        assert(self->ancestor_map[input_id] != NULL);
```

Comment:
I moved these `assert`s up top from down below since they do not depend on the processing of the edges.